src/chains/ethereum/ethereum/src/miner/miner.ts

```diff
@@ -370,7 +370,7 @@ export default class Miner extends Emittery<{
             // notice: when `maxTransactions` is `-1` (AKA infinite), `numTransactions === maxTransactions`
             // will always return false, so this comparison works out fine.
             if (
-              blockGasLeft <= Params.TRANSACTION_GAS ||
+              blockGasLeft < Params.TRANSACTION_GAS ||
```

Comment:
Writing my tests lead me to this bug - if the block's remaining gas was _equal_ to the amount of gas it takes to run an empty transaction, we'd break out. If a block had used `gasLimit - 21000`, it wouldn't mine one more empty transaction. This can be another PR if needed.